clippy_lints/src/manual_and.rs

```diff
@@ -0,0 +1,124 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::visitors::for_each_expr;
+use clippy_utils::{get_parent_expr, peel_blocks};
+use core::ops::ControlFlow;
+use rustc_ast::ast::LitKind;
+use rustc_ast::BinOpKind;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects `if`-then-`else` that can be replaced with `&&`.
+    ///
+    /// ### Why is this bad?
+    /// `&&` is simpler than `if`-then-`else`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// if a {
+    ///     b
```

Comment:
doc test is failling, maybe you meant to use `/// ```ignore` ?. Or assign values to `a` and `b` using:

```rust
/// ```no_run
/// # let a = true
/// # let b = false
/// if a {
///     b
```

something like that~